skimage/feature/peak.py

```diff
@@ -22,6 +22,40 @@ def _get_high_intensity_peaks(image, mask, num_peaks):
     return coord[::-1]
 
 
+def _get_peak_mask(image, min_distance, footprint, threshold_abs,
+                   threshold_rel):
+    """
+    Return the mask containing all peak candidates above thresholds.
+    """
+    if footprint is not None:
+        image_max = ndi.maximum_filter(image, footprint=footprint,
+                                       mode='constant')
+    else:
+        size = 2 * min_distance + 1
+        image_max = ndi.maximum_filter(image, size=size, mode='constant')
+    mask = image == image_max
+    if threshold_rel is not None:
+        threshold = max(threshold_abs, threshold_rel * image.max())
+    else:
+        threshold = threshold_abs
+    mask &= image > threshold
+    return mask
+
+
+def _exclude_border(mask, footprint, exclude_border):
+    """
+    Remove peaks round the borders.
+    """
+    # zero out the image borders
+    for i in range(mask.ndim):
+        mask = mask.swapaxes(0, i)
+        remove = (footprint.shape[i] if footprint is not None
+                  else 2 * exclude_border)
+        mask[:remove // 2] = mask[-remove // 2:] = False
```

Comment:
I really dislike swapping axes

```
mask[(slice(None),) * i + slice(None, remove // 2)] = False
mask[(slice(None),) * i + slice(-remove // 2, None)] = False
```

Reply:
Sorry, are you referring to the line above or below or both?

Reply:
Replace 
```
        mask = mask.swapaxes(0, i)
        remove = (footprint.shape[i] if footprint is not None
                  else 2 * exclude_border)
        mask[:remove // 2] = mask[-remove // 2:] = False
        mask = mask.swapaxes(0, i)

```
with what @hmaarrfk provided:

```
        remove = (footprint.shape[i] if footprint is not None
                  else 2 * exclude_border)
        mask[(slice(None),) * i + slice(None, remove // 2)] = False
        mask[(slice(None),) * i + slice(-remove // 2, None)] = False
```

Reply:
The code actually should be:
```
def _exclude_border(mask, footprint, exclude_border):
    """
    Remove peaks round the borders
    """
    # zero out the image borders
    for i in range(mask.ndim):
        remove = (footprint.shape[i] if footprint is not None
                  else 2 * exclude_border)
        mask[(slice(None),) * i + (slice(None, remove // 2),)] = False
        mask[(slice(None),) * i + (slice(-remove // 2, None),)] = False
    return mask
```

Reply:
Thanks for making that super explicit

Reply:
The comment should be ("round" was a typo):

Remove peaks near the borders